sim/src/main/cc/bridges/UARTModule.cc

```diff
@@ -0,0 +1,75 @@
+// See LICENSE for license details.
+
+#ifndef RTLSIM
+#include "simif_f1.h"
+#define SIMIF simif_f1_t
+#else
+#include "simif_emul.h"
+#define SIMIF simif_emul_t
+#endif
+
+#include "bridges/uart.h"
+#include "simif_peek_poke.h"
+
+class UARTModuleTest final : public simif_peek_poke_t {
+public:
+  class Handler final : public uart_handler {
+  public:
+    ~Handler() {
+      // Check that the input and output buffers are equal.
+      if (in_buffer != out_buffer) {
+        fprintf(stderr,
```

Comment:
I'm curious, what's your opinion on streams vs printf for output to stderr?

Reply:
I've used printf since it was used in other parts as well. But overall I would prefer streams.

Moving forward, I would like to move the test checking functions out of `simif_peek_poke_t` into `TestHarness`, adding googletest-like assertions there as well for verification, so there is no need to print anything in tests.

Reply:
Yeah, that sounds ideal. Thanks for the context.